apps/src/applab/applab.js

```diff
@@ -510,21 +508,31 @@ Applab.getCode = function () {
 Applab.getHtml = function () {
   // This method is called on autosave. If we're about to autosave, let's update
   // levelHtml to include our current state.
-  if (Applab.isInDesignMode() && !Applab.isRunning() || Applab.levelHtml === '') {
+  if ($('#designModeViz').is(':visible')) {
     designMode.serializeToLevelHtml();
   }
   return Applab.levelHtml;
 };
 
+/**
+ * Sets Applab.levelHtml as well as #designModeViz contents.
+ * designModeViz is the source of truth for the app's HTML.
+ * levelHtml can be lazily updated from designModeViz via serializeToLevelHtml.
+ * @param html
+ */
 Applab.setLevelHtml = function (html) {
   if (html === '') {
-    if (window.dashboard) {
-      dashboard.project.clearHtml();
-    }
     Applab.levelHtml = '';
   } else {
     Applab.levelHtml = designMode.addScreenIfNecessary(html);
   }
+  var designModeViz = document.getElementById('designModeViz');
+  designMode.parseFromLevelHtml(designModeViz, true);
```

Comment:
Yeah, our code has lots of unidentifiable params being passed to functions. I've also seen these two conventions:
```
designMode.parseFromLevelHtml(designModeViz, true /* allowDragging */ );
```
or
```
var allowDragging = true;
designMode.parseFromLevelHtml(designModeViz, allowDragging);
```
However (until we do ES6) I think I like the options params hash better since once you define the function that way, the caller is forced to make their call readable.